project2/screens/search.js

```diff
@@ -1,131 +1,72 @@
 import React from "react";
-import { Button, FlatList, StyleSheet, Text, TouchableOpacity, TextInput, View } from "react-native";
-import Constants from 'expo-constants'
+import { FlatList, Text, TextInput, View } from "react-native";
 import { searchMovies, getMoviePage } from "../API";
-
-
-const Row = props => (
-    <TouchableOpacity onPress={() => props.selected(props.id)}>
-        <View style={styles.row}>
-            <Text style={styles.listText}>{props.title}</Text>
-        </View>
-    </TouchableOpacity>
-)
-const renderItem = selected => obj => <Row selected={selected}{...obj.item} />
-
-export default class search extends React.Component {
-    static navigationOptions = {
-        title: "Search",
-        headerTitleStyle: { color: "#AAAAAF", }
-    };
-
-    state = {
-        searchString: "",
-        movies: "",
-        totalResults: 0,
-        totalPages: 0,
-        nextPage: 0,
-    };
-
-    getMovies = async (searchString) => {
-        const results = await searchMovies(`${searchString}`)
-        console.log(results)
-
-        if (results !== undefined) {
-            this.setState({ movies: results["result"] })
-            this.setState({ totalResults: results["totalResults"] })
-            this.setState({ totalPages: results["totalPages"] })
-
-            if (results["totalPages"] > 1) {
-                this.setState({ nextPage: 2 })
-            }
-        }
-    }
-
-    componentDidUpdate(prevProps, prevState) {
-        if (this.state.searchString !== prevState.searchString) {
-            this.getMovies(this.state.searchString)
+import SearchListItem from "./Search/ListItem";
+import s from "../Styles";
+
+const Search = () => {
+  const [searchString, setSearchString] = React.useState("");
+  const [movies, setMovies] = React.useState([]);
+  const [totalResults, setTotalResults] = React.useState(0);
+  const [totalPages, setTotalPages] = React.useState(0);
+  const [nextPage, setNextPage] = React.useState(0);
```

Comment:
Multiple `useState`s can be used, and it is preferable to do this (there is a very good alternative, `useReducer`, for complex state like this, but `useState` is conceptually much simpler). Because a component rerenders every time state changes, you want to be as granular as possible, and if at all possible have individual bits of state as primitive values (which can be diff-ed for changes easily) rather than objects (which cannot be easily diff-ed for changes, as they are references, not values).